### PR TITLE
Include Eth1 RPC Endpoint Mock Server

### PR DESCRIPTION
**Pull request by rauljordan**
This PR implements an ETH1 Mock RPC server that can serve as a plug-in endpoint into an eth2 node, providing the node with proper information to reach genesis, handle new validator deposits, and call methods from the deposit contract. Instead of connecting to a real network, the server uses mocks to properly respond with correct data that would require zero changes to the eth2 client.

### CODE REVIEW

File: json.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
+// Copyright 2015 The go-ethereum Authors
```
Comment (rauljordan) [inline json.go]: File taken straight from the go-ethereum repository - do not review

File: websocket.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The go-ethereum Authors
```
Comment (rauljordan) [inline websocket.go]: Logic taken directly from the go-ethereum repository - do not review

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (prestonvanloon) [inline LICENSE]: Fill this in?

File: json.go (hunk 1 above)
Comment (prestonvanloon) [reply json.go]: Why not use this package from upstream? Why does it need to be checked in

File: websocket.go (hunk 2 above)
Comment (prestonvanloon) [reply websocket.go]: Likewise here, why is this copied here instead of used from upstream?

File: json.go (hunk 1 above)
Comment (rauljordan) [reply json.go]: I need a lot of its internals that are unexported variables and methods

File: websocket.go (hunk 2 above)
Comment (rauljordan) [reply websocket.go]: Unexported in upstream

**Pull request merged**